docs/metrics/custom-metrics.md

```diff
@@ -0,0 +1,122 @@
+---
+template: main.html
+---
+
+# Custom Metrics
+
+Custom Iter8 metrics enable you to use data from any database for evaluating app/ML model versions within Iter8 experiments. This document describes how you can define custom Iter8 metrics and (optionally) supply authentication information that may be required by the metrics provider.
+
+Metric providers differ in the following aspects.
+
+* HTTP request authentication method: no authentication, basic auth, API keys, or bearer token
+* HTTP request method: GET or POST
+* Format of HTTP parameters and/or JSON body used while querying them
+* Format of the JSON response returned by the provider
+* The logic used by Iter8 to extract the metric value from the JSON response
+
+The examples in this document focus on Prometheus, NewRelic, Sysdig, and Elastic. However, the principles illustrated here will enable you to use metrics from any provider in experiments.
+
+## Metrics with/without auth
+
+> **Note:** Metrics are defined by you, the **Iter8 end-user**.
+
+=== "Prometheus"
+
+    Prometheus does not support any authentication mechanism *out-of-the-box*. However,
```

Comment:
Has this changed? The Prometheus documentation suggests that it is supported. https://prometheus.io/docs/prometheus/latest/configuration/https/#https-and-authentication

Reply:
Only as an `experimental` feature. And they are saying this might change (which includes dropping the feature altogether).

Reply:
@kalantar @sriumcp What is the conclusion here? Do I need to change anything? I can also take this out for now.

Reply:
I think it is fine as it is, at least for now. We can revisit later.